src/RuleCollection.php

```diff
@@ -862,21 +862,28 @@ public function moveRule($ID, $ref_ID, $type = self::MOVE_AFTER)
         $ruleDescription->getFromDB($ID);
         $old_rank = $ruleDescription->fields["ranking"];
 
-       // Compute new ranking
-        if ($ref_ID) { // Move after/before an existing rule
-            $ruleDescription->getFromDB($ref_ID);
-            $rank = $ruleDescription->fields["ranking"];
-        } else if ($type == self::MOVE_AFTER) {
-           // Move after all
-            $result = $DB->request([
-                'SELECT' => ['MAX' => 'ranking AS maxi'],
-                'FROM'   => 'glpi_rules',
-                'WHERE'  => ['sub_type' => $this->getRuleClassName()]
-            ])->current();
-            $rank   = $result['maxi'];
+        $max_ranking_criteria = [
```

Comment:
I can't understand the changes below.
And you don't seems to use them in the current PR.
Do I miss something ?

Reply:
It has been a while since I made this PR, but I seem to remember making this change to make it possible to specify the new rank directly via the API without having duplicates. Having to make the rank an object during create/update actions so it can take a reference ID and a type (before/after) would have been complicated and potentially confusing.

Indeed, I don't see anything calling this method, so I will test it later.